hyperstart/api/grpc/utils.go

```diff
@@ -0,0 +1,140 @@
+package grpc
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func copyValue(to, from reflect.Value) error {
+	toKind := to.Kind()
+	fromKind := from.Kind()
+
+	if !from.IsValid() {
+		return nil
+	}
+
+	if toKind == reflect.Ptr {
+		// If the destination is a pointer, we need to allocate a new one.
+		to.Set(reflect.New(to.Type().Elem()))
+		if fromKind == reflect.Ptr {
+			return copyValue(to.Elem(), from.Elem())
+		} else {
+			return copyValue(to.Elem(), from)
+		}
+	} else {
+		// Here the destination is not a pointer.
+		// Let's check what's the origin.
+		if fromKind == reflect.Ptr {
+			return copyValue(to, from.Elem())
+		}
+
+		switch toKind {
+		case reflect.Struct:
+			return copyStructValue(to, from)
+		case reflect.Slice:
```

Comment:
Do you need to handle arrays as well?  They have a different reflection type.

Reply:
@markdryan We do not need to support arrays for now.